### PR TITLE
COBRA-3058: add "pipeline_promotion" hook

### PR DESCRIPTION
**Pull request by sbeck14**
Send a "pipeline_promotion" hook on all target apps in a pipeline coupling upon successful promotion, and include information about the source app and pending release.

### CODE REVIEW

File: docs/API-Reference.md [hunk 1]
```diff
@@ -5228,6 +5229,42 @@ This event occurs when an app has a new version and is now available for request
...
+  },
+  "release":{
+    "id":"1edbac4b-6a5e-09e1-ef3a-08084a904623",
+    "status":"queued",
+    "created_at":"2016-08-09T12:00:00Z",
+    "version":13,
+    "description":"Promotion from sourceapp-source-space"
+  }
+}
+```
```
Comment (trevorlinton) [inline docs/API-Reference.md]: At the top of this doc section there's a list of all of the events, can you add this new event to that list?

File: lib/pipelines.js [hunk 2]
```diff
@@ -476,6 +476,32 @@ async function http_create_pipeline_promotion(pg_pool, req, res, regex) {
...
+          'name:': app.app_name,
+          'id': app.app_uuid,
+        },
+        'space': {
+          'name': app.space_name,
+        },
+      },
+      'release': {
+        'id': release.release_id,
+        'build': release.build_uuid,
```
Comment (trevorlinton) [inline lib/pipelines.js]: The documentation does not mention that the build uuid is included, in addition, if we keep the build id can we pull it out from the release object and have `"build":{"id":release.build_uuid}` off of the root? Reason why, is if we decide to add more information about the build (such as the sha, repo, or commit message or something else) we don't have to modify or remove properties from the structure of the payload. Makes it easier to add new information to payloads without breaking everyone.

File: docs/API-Reference.md [hunk 3]
```diff
@@ -4766,10 +4766,11 @@ The following events exist for hooks to listen to:
 * preview-released
 * released
 * crashed
+* pipeline_promotion
 
-When a hook is called the URL provided is called with a `POST` method, the body is different depending on the event but will always have the property "action" that equals the event name. 
+When a hook is called, the URL provided is called with a `POST` method. While the body of the `POST` may differ slightly depending on the event, it will always have the property "action" that equals the event name. 
 
-Finally you can rely on the following headers being available on each of the webhook calls:
+Finally, you can rely on the following headers being available on each of the webhook calls:
```
Comment (trevorlinton) [inline docs/API-Reference.md]: Thanks for fixing my wonky grammar 😄

File: docs/API-Reference.md (hunk 1 above)
Comment (trevorlinton) [reply docs/API-Reference.md]: Nevermind, I just realized you did. Sorry

File: lib/pipelines.js (hunk 2 above)
Comment (sbeck14) [reply lib/pipelines.js]: Ah yeah I gotcha. Will do

**Pull request merged**